### ISSUE TITLE
Pre-training speed is slow.

### ISSUE THREAD

**Issue by wangtianbaowtb** [jayleicn/ClipBERT#28]
Thank you for releasing the code.
We tried to reproduce the pre-training work, using 8 v100 and the parameter settings are the same as in the paper, batch_size=32, num_workers=4, but the training speed is always very slow, the GPU utilization rate vibrates between 0 and 100% , most of the time is 0, the CPU occupancy rate is about 30%. We need about 350 hours to train for 40 epochs, which is about 4 times the amount mentioned in the paper. We think maybe the dataloader is the bottleneck , but the training speed  has not improved when we using larger num_workers like 8. On the other hand, when we only use one GPU for training, the GPU utilization can always reach 100%, and the total time is only about 370 hours.
So we would like to ask whether there is anything wrong in our pre-training work and what is the possible reason. Thank you and looking forward to a reply.

**Comment by jayleicn** [comment]
Hi @wangtianbaowtb, Are you using HDD drive? This is probably related to the disk loading speed?

**Comment by wangtianbaowtb** [comment]
Thank you for timely response.
Yes, due to the large scale of the dataset, we store the data on HDD. We also encountered the same problem when we reproducing your benchmark work HERO, so we tried to store the HERO's vqa dataset "tvqa" in tmpfs to test whether the training speed was slow due to dataloading, but we found it has little effect on training speed, wherever the dataset is in HDD or tmpfs. We would like to ask, do you store all the required datasets in the SSD? Has your gpu utilization rate been maintained at a high level? Do you think there is a problem with backpropagation that causes the training speed to be slow? Thank you very much！

**Comment by jayleicn** [comment]
HERO is slow mainly (probably?) due to its relatively complex model architecture, e.g.,  it may require padding at different levels of hierarchy. Since it loads features and raw text directly, without any additional steps, data loading should not be a probelm.

For ClipBERT, it loads raw images/videos and requires performing some heavy decoding steps, etc. Both disk data loading and cpu decoding could be bottleneck here.
